vortex-ffi/src/file.rs

```diff
@@ -35,6 +36,23 @@ arc_wrapper!(
     vx_file
 );
 
+fn replace_file_idx(file_idx: u64, expr: ExprRef) -> VortexResult<ExprRef> {
+    expr.transform(&mut pre_order_mut_visit_up(|e: ExprRef| {
+        if let Some(get_item) = e.as_any().downcast_ref::<GetItem>() {
+            if get_item.field().as_ref() != "file_index" {
+                return Ok(TransformResult::no(e));
+            }
+            if let Some(var) = get_item.as_any().downcast_ref::<Var>() {
+                if var.var() == "$vx.row_id" {
+                    return Ok(TransformResult::yes(lit(file_idx)));
+                }
+            }
+        }
+        return Ok(TransformResult::no(e));
+    }))
+    .map(|r| r.into_inner())
```

Comment:
I feel like this should just be a new variable in the scope, right?

Reply:
yes, in the future, however we don't have the APIs for that now, right?